src/sd.jl

```diff
@@ -0,0 +1,26 @@
+export PSDCone
+# Used in @constraint m X in SDP
+struct PSDCone end
+
+struct SDVariableConstraint <: AbstractConstraint
+    Q::Matrix{JuMP.Variable}
+end
+
+# Used by the @variable macro but can also be used through the @constraint macro
+function constructconstraint!(Q::Matrix{JuMP.Variable}, PSDCone)
+    @assert issymmetric(Q) # TODO it could be nonsymmetric if used through the @constraint macro
```

Comment:
We're not currently constructing variablewise constraints via the `@constraint` macro. This needs a larger discussion if we're going to allow this syntax.